src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -642,6 +658,11 @@ private void PopulateProjectAndPackageReferences(PackageBuilder packageBuilder,
                         continue;
                     }
 
+                    var versionToUse = targetLibrary.Version;
+                    if(!readProjectRefVersionFromAssetsFile)
+                    {
+                        versionToUse = NuGetVersion.Parse(projectRefToVersionMap[projectReference.ProjectPath]);
```

Comment:
what if it failed to parse the NuGet version?

Reply:
the pack task has error handling at the entry point, so whatever exception it throws will be gracefully displayed on the console